components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream_test.go

```diff
@@ -0,0 +1,154 @@
+package jetstream
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/subscriptionmanager"
+	controllertesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type jetStreamSubMgrMock struct {
+	Client  dynamic.Interface
+	Backend handlers.JetStreamBackend
+}
+
+func (c *jetStreamSubMgrMock) Init(_ manager.Manager) error {
+	return nil
+}
+
+func (c *jetStreamSubMgrMock) Start(_ env.DefaultSubscriptionConfig, _ subscriptionmanager.Params) error {
+	return nil
+}
+
+func (c *jetStreamSubMgrMock) Stop(_ bool) error {
+	return nil
+}
+
+func TestCleanup(t *testing.T) {
```

Comment:
The test is really long. Can it be split up into smaller and reusable functions ? For example refactoring out the actual test setup as in the [guidelines](https://github.com/kyma-project/community/blob/main/concepts/eventing-code-guidelines.md#separate-test-setup-from-actual-test).

It all boils down to something like this:

```
	// given
	// init something
	
	// when
	err = cleanup(jsSubMgr.Backend, jsSubMgr.Client, defaultLogger.WithContext())

	// then
	// ensure jetstream subscription is deleted
```

Reply:
I'll try to cleanup👍